ang/civicase/ContactCaseTab.js

```diff
@@ -0,0 +1,215 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase');
+
+  module.directive('civicaseContactCaseTab', function () {
+    return {
+      restrict: 'EA',
+      controller: CivicaseContactCaseTabController,
+      templateUrl: '~/civicase/ContactCaseTab.html',
+      scope: {}
+    };
+  });
+
+  function CivicaseContactCaseTabController ($scope, crmApi, formatCase) {
+    $scope.casesList = [
+      {
+        'name': 'opened',
+        'title': 'Open Cases',
+        'filterParams': {
+          'status_id.grouping': 'Opened',
+          'contact_id': getSelectedContactId()
+        },
+        'isLoaded': false,
+        'showSpinner': false,
+        'cases': [],
+        'isLoadMoreAvailable': false,
+        'showContactRole': false,
+        'page': {
+          'size': 3,
+          'num': 1
+        }
+      }, {
+        'name': 'closed',
+        'title': 'Closed Case',
+        'filterParams': {
+          'status_id.grouping': 'Closed',
+          'contact_id': getSelectedContactId()
+        },
+        'isLoaded': false,
+        'showSpinner': false,
+        'cases': [],
+        'isLoadMoreAvailable': false,
+        'showContactRole': false,
+        'page': {
+          'size': 3,
+          'num': 1
+        }
+      }, {
+        'name': 'related',
+        'title': 'Other cases for this contact',
+        'filterParams': {
+          'case_manager': getSelectedContactId()
+        },
+        'isLoaded': false,
+        'showSpinner': false,
+        'cases': [],
+        'isLoadMoreAvailable': false,
+        'showContactRole': true,
+        'page': {
+          'size': 3,
+          'num': 1
+        }
+      }
+    ];
+
+    $scope.checkPerm = CRM.checkPerm;
+    $scope.ts = CRM.ts('civicase');
+
+    (function init () {
+      initSubscribers();
+      getCases();
+    }());
+
+    /**
+     * refresh function to set refresh cases
+     */
+    $scope.refresh = function () {
+      $scope.$emit('civicase::contact-record-case::refresh-cases');
```

Comment:
Which comp is using this?

Reply:
Fixed the logic. Apologies to send the broken implementation for review.